miden-lib/asm/sat/note_setup.masm

```diff
@@ -0,0 +1,38 @@
+use.miden::sat::prologue
+
+# MEMORY OFFSETS
+# =================================================================================================
+
+# BOOK KEEPING
+# -------------------------------------------------------------------------------------------------
+const.CURRENT_CONSUMED_NOTE_PTR=3
+const.CONSUMED_NOTE_INPUTS_HASH_OFFSET=3
+
+# NOTE SETUP SCRIPT
+# =================================================================================================
+
+export.note_setup_script.5
+    # convert the consumed note index to a pointer and store in memory
+    exec.prologue::consumed_note_data_ptr push.CURRENT_CONSUMED_NOTE_PTR mem_store
+
+    # load the note inputs from the advice provider
+    locaddr.0 padw padw padw adv_pipe adv_pipe
+
+    # extract inputs hash and assert it matches commitment stored in memory
+    dropw swapw dropw
+    padw push.CURRENT_CONSUMED_NOTE_PTR mem_load
+    push.CONSUMED_NOTE_INPUTS_HASH_OFFSET add mem_loadw
```

Comment:
minor nit (feel free to ignore): maybe encapsulate this in a separate procedure? Something like `get_current_note_inputs_hash`?